src/index.ts

```diff
@@ -46,217 +47,222 @@ app.use((req, res, next) => {
+            headerOvalAddresses,
+          );
+
+          const { unlockSignedTransactions, unlockTxHashes } = await getUnlockBundlesFromOvalAddresses(
+            flashbotsBundleProvider,
+            backrunTxs,
+            targetBlock,
+            headerOvalAddresses,
+            req,
+          );
+          simulationResponse = await flashbotsBundleProvider.simulate(
+            [...unlockSignedTransactions, ...backrunTxs],
+            targetBlock,
+          );
+          unlockTransactionHashes = unlockTxHashes;
+        } else {
+          Logger.debug(req.transactionId, "Finding unlock that does not revert the bundle...");
+
+          const unlock = await findUnlock(flashbotsBundleProvider, backrunTxs, targetBlock, req);
+          if (!unlock) {
+            Logger.debug(req.transactionId, "No valid unlock found!");
+            await handleUnsupportedRequest(req, res, "No valid unlock found"); // Pass through if no unlock is found.
+            return;
+          }
 
-        Logger.debug(
-          req.transactionId,
-          `Found valid unlock at ${unlock.ovalAddress}. Simulating unlock tx bundle and backrun bundle...`,
-        );
+          Logger.debug(
+            req.transactionId,
+            `Found valid unlock at ${unlock.ovalAddress}. Simulating unlock tx bundle and backrun bundle...`,
+          );
 
-        simulationResponse = await flashbotsBundleProvider.simulate(
-          [unlock.signedUnlockTx, ...backrunTxs],
-          targetBlock,
-        );
-        unlockTransactionHashes = [unlock.unlockTxHash];
-      }
+          simulationResponse = await flashbotsBundleProvider.simulate(
+            [unlock.signedUnlockTx, ...backrunTxs],
+            targetBlock,
+          );
+          unlockTransactionHashes = [unlock.unlockTxHash];
+        }
 
-      // Send back the simulation response without the unlock transaction.
-      return handleBundleSimulation(simulationResponse, unlockTransactionHashes, req, res);
-    } else await handleUnsupportedRequest(req, res, "Invalid signature or method");
-  } catch (error) {
-    next(error);
-  }
-});
+        // Send back the simulation response without the unlock transaction.
+        return handleBundleSimulation(simulationResponse, unlockTransactionHashes, req, res);
+      } else await handleUnsupportedRequest(req, res, "Invalid signature or method");
+    } catch (error) {
+      next(error);
+    }
+  });
 
-// Health check endpoint.
-app.get("/ready", (req, res) => {
-  res.status(200).send("OK");
-});
+  // Health check endpoint.
+  app.get("/ready", (req, res) => {
+    res.status(200).send("OK");
+  });
 
-app.use(expressErrorHandler);
+  app.use(expressErrorHandler);
 
-app.listen(env.port, () => {
-  Logger.debug("Startup", `Server is running on port ${env.port}`);
-});
+  app.listen(env.port, () => {
+    Logger.debug("Startup", `Server is running on port ${env.port}`);
+  });
+})();
```

Comment:
it looks like this cant actually throw, but if it can this needs to use a .catch or we have unhandled async error